File: Emulator/src/main/content/global/skill/summoning/familiar/npc/SpiritKyattNPC.java
package content.global.skill.summoning.familiar.npc;

import content.global.skill.summoning.familiar.Familiar;
import content.global.skill.summoning.familiar.FamiliarSpecial;
import core.game.node.entity.combat.equipment.WeaponInterface;
import core.game.node.entity.player.Player;
import core.game.node.entity.skill.SkillBonus;
import core.game.node.entity.skill.Skills;
import core.plugin.Initializable;

/**
 * The type Spirit kyatt npc.
 */
@Initializable
public class SpiritKyattNPC extends content.global.skill.summoning.familiar.Familiar {

    /**
     * Instantiates a new Spirit kyatt npc.
     */
    public SpiritKyattNPC() {
        this(null, 7366);
    }

    /**
     * Instantiates a new Spirit kyatt npc.
     *
     * @param owner the owner
     * @param id    the id
     */
    public SpiritKyattNPC(Player owner, int id) {
        super(owner, id, 4900, 12812, 3, WeaponInterface.STYLE_ACCURATE);
        boosts.add(new SkillBonus(Skills.HUNTER, 5));
    }

    @Override
    public Familiar construct(Player owner, int id) {
        return new SpiritKyattNPC(owner, id);
    }

    @Override
    protected boolean specialMove(FamiliarSpecial special) {
        if (!super.isOwnerAttackable()) {
            return false;
        }
        call();
        return true;
    }

    @Override
    public int[] getIds() {
        return new int[]{7365, 7366};
    }

}

File: Server/src/main/content/global/skill/summoning/familiar/npc/SpiritKyattNPC.kt
package content.global.skill.summoning.familiar.npc

import content.global.skill.summoning.familiar.Familiar
import content.global.skill.summoning.familiar.FamiliarSpecial
import core.game.node.entity.combat.equipment.WeaponInterface
import core.game.node.entity.player.Player
import core.game.node.entity.skill.SkillBonus
import core.game.node.entity.skill.Skills
import core.plugin.Initializable
import shared.consts.Items
import shared.consts.NPCs

@Initializable
class SpiritKyattNPC @JvmOverloads constructor(owner: Player? = null, id: Int = NPCs.SPIRIT_KYATT_7366) :
    Familiar(owner, id, 4900, Items.SPIRIT_KYATT_POUCH_12812, 3, WeaponInterface.STYLE_ACCURATE) {

    init {
        boosts.add(SkillBonus(Skills.HUNTER, 5.0))
    }

    override fun construct(owner: Player, id: Int): Familiar {
        return SpiritKyattNPC(owner, id)
    }

    override fun specialMove(special: FamiliarSpecial): Boolean {
        if (!super.isOwnerAttackable()) {
            return false
        }
        call()
        return true
    }

    override fun getIds(): IntArray {
        return intArrayOf(NPCs.SPIRIT_KYATT_7365, NPCs.SPIRIT_KYATT_7366)
    }
}
